tools/rabbitmq-consumer/src/RabbitMQConsumer/Lib.hs

```diff
@@ -0,0 +1,203 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2023 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+{-# LANGUAGE OverloadedStrings #-}
+
+module RabbitMQConsumer.Lib where
+
+import Data.Aeson (FromJSON, decode)
+import Data.ByteString.Lazy.Char8 qualified as BL
+import Imports
+import Network.AMQP
+import Network.Socket
+import Options.Applicative
+
+main :: IO ()
+main = do
+  opts <- execParser (info (helper <*> optsParser) desc)
+  conn <- openConnection' opts.host opts.port opts.vhost opts.username opts.password
+  chan <- openChannel conn
+  qos chan 0 1 False
+  done <- newEmptyMVar
+  runTimerAsync done opts.timeoutSec
+  case opts.cmd of
+    Head -> void $ consumeMsgs chan opts.queue Ack (printHead done opts)
+    Interactive -> void $ consumeMsgs chan opts.queue Ack (interactive done opts)
+    DropHead dhOpts -> void $ consumeMsgs chan opts.queue Ack (dropHead done opts dhOpts)
+  takeMVar done
```

Comment:
Seems wrong to run timer like this for interactive session. Things could timeout when the user is just looking at the JSON.